eng/code-quality-reports/src/main/resources/revapi/revapi.json

```diff
@@ -105,6 +113,54 @@
     "configuration": {
       "ignore": true,
       "differences": [
+        {
+          "regex": true,
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "(class|interface|enum) org.springframework..*",
+          "justification": "springframework classes are allowed to be exposed."
+        },
+        {
+          "regex": true,
+          "code": "java.class.nonPublicPartOfAPI",
+          "new": "(class|interface) org.springframework..*",
+          "justification": "We allow this in springframework.cloud."
+        },
+        {
+          "regex": true,
+          "code": "java.method.addedToInterface",
+          "new": "method <A extends java.lang.annotation.Annotation> A org.springframework.beans.factory.ListableBeanFactory::findAnnotationOnBean()",
+          "justification": "springframework are allowed to add methods.."
+        },
+        {
+          "regex": true,
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "(class|interface|enum) com.azure.messaging.(servicebus|eventhubs)..*",
+          "justification": "azure.messaging classes are allowed to be exposed."
+        },
+        {
+          "regex": true,
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "class com.azure.data.appconfiguration..*",
+          "justification": "azure.data.appconfiguration classes are allowed to be exposed."
+        },
+        {
+          "regex": true,
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "class com.azure.identity..*",
+          "justification": "azure.identity classes are allowed to be exposed."
+        },
+        {
+          "regex": true,
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "(class|interface|enum|method) com.azure.spring..*",
+          "justification": "azure.spring classes are allowed to be exposed."
+        },
+        {
+          "regex": true,
+          "code": "java.class.nonPublicPartOfAPI",
+          "new": "class com.azure.spring.messaging.container.DefaultMessageListenerContainer",
+          "justification": "We allow this in azure.spring.messaging."
+        },
```

Comment:
Why do we need this?

Reply:
Okay, I figured it out. The visibility of `DefaultMessageListenerContainer` is package level but it's returned from the `AbstractAzureListenerContainerFactory.createListenerContainer` method.